### PR TITLE
Fix querying datapoints within a day

### PR DESCRIPTION
**Pull request by emelvarol-cognite**
CDF-16612

### CODE REVIEW

File: src/test/scala/cognite/spark/v1/DataPointsRelationTest.scala [hunk 1]
```diff
@@ -1059,4 +1061,92 @@ class DataPointsRelationTest extends FlatSpec with Matchers with ParallelTestExe
...
+
+    val res3 = spark.sql(
+      """SELECT dp.timestamp, dp.value FROM destinationDatapointsBluefield dp
+        |WHERE dp.externalId == 'emel' AND
+        |dp.timestamp > TO_TIMESTAMP('2022-09-01T00:00:00Z') AND
+        |dp.timestamp <= TO_TIMESTAMP('2022-09-02T00:00:00Z')""".stripMargin).collect()
+    res3.length shouldBe 3
+    res3.map(row => row.getDouble(1)).toSet shouldBe Set(0.3, 0.4, 0.5)
+
+    val res4 = spark.sql(
```
Comment (emelvarol-cognite) [inline src/test/scala/cognite/spark/v1/DataPointsRelationTest.scala:1129]: this doesnt work

File: src/main/scala/cognite/spark/v1/NumericDataPointsRdd.scala [hunk 2]
```diff
@@ -244,24 +244,41 @@ final case class NumericDataPointsRdd(
...
       ids: Vector[CogniteId],
       start: Instant,
-      end: Instant): IO[Vector[(CogniteId, Option[Instant], Option[Instant])]] =
+      end: Instant): IO[Vector[(CogniteId, Option[Instant], Option[Instant])]] = {
+    // Filter conversions before this function assumes end limit is inclusive in data points, we add 1 milliseconds
+    // since it is exclusive in the API
+    val exclusiveEnd = end.plusMillis(1)
     for {
       // fetch firsts before lasts since that can correctly handle when accidentally reading
       // stringdatapoints with a reasonable error message
```
Comment (emelvarol-cognite) [inline src/main/scala/cognite/spark/v1/NumericDataPointsRdd.scala:253]: Haha, I will fix the comments sorry 🤦

**Review by vuhainguyen — APPROVED** [review]
❤️ the tests that covers different scenarios and edge cases 😁 
It would be great if we can have a basic setup for oidc migration with those bluefield variables :)
Please update changeLog and bump version.  (Can you bump Spark version to 3.3.1 as well ?, I already merged [this](https://github.com/cognitedata/docker-spark/pull/69)
